ledger_device_sdk/src/io.rs

```diff
@@ -268,7 +280,80 @@ impl Comm {
         }
     }
 
-    pub fn decode_event<T>(&mut self, spi_buffer: &mut [u8; 128]) -> Option<Event<T>>
+    pub fn next_event_ahead<T>(&mut self) -> bool
+    where
+        T: TryFrom<ApduHeader>,
+        Reply: From<<T as TryFrom<ApduHeader>>::Error>,
+    {
+        // if self.event_pending {
```

Comment:
It seems like this is not needed... it works with or without it... Not sure why @kingofpayne

Reply:
See previous response related to this.
Maybe the successive calls to `seph_recv` silently fail or do nothing because you can't receive a new event as you did not reply to the pending APDU. If this is the case then the apdu buffer remains unmodified and still holds the previous APDU, so `detect_apdu` would still return `true`.